src/modules/75-cf/components/FlagArchiving/__tests__/useArchiveFlagDialog.Git.test.tsx

```diff
@@ -0,0 +1,124 @@
+import mockGitSync from '@cf/utils/testData/data/mockGitSync'
+import useArchiveFlagDialog, { ArchiveDialogProps } from '../useArchiveFlagDialog'
+
+const queryParamsMock = {
+  accountIdentifier: 'mockAccountIdentifier',
+  orgIdentifier: 'mockOrgIdentifier',
+  projectIdentifier: 'mockProjectIdentifier',
+  forceDelete: false
+}
+
+const openArchiveDialogBtn = 'Open Archive dialog'
+
+const WrapperComponent: FC<ArchiveDialogProps> = ({
+  flagData,
+  archiveFlag,
+  onArchive,
+  queryParams,
+  gitSync,
+  openedArchivedDialog
+}) => {
+  const { openDialog } = useArchiveFlagDialog({
+    flagData,
+    archiveFlag,
+    onArchive,
+    queryParams,
+    gitSync,
+    openedArchivedDialog
+  })
+
+  return <button onClick={() => openDialog()}>{openArchiveDialogBtn}</button>
+}
+
+const renderComponent = (props: Partial<ArchiveDialogProps> = {}): RenderResult => {
+  return render(
+    <TestWrapper>
+      <FFGitSyncProvider>
+        <WrapperComponent
+          archiveFlag={jest.fn()}
+          flagData={mockFeature as Feature}
+          gitSync={mockGitSync}
+          onArchive={jest.fn()}
+          queryParams={queryParamsMock}
+          openedArchivedDialog={false}
+          {...props}
+        />
+      </FFGitSyncProvider>
+    </TestWrapper>
+  )
+}
+
+const setUseGitRepoMock = (repoDetails: Partial<GitRepo> = {}, repoSet = true): void => {
+  jest.spyOn(cfServices, 'useGetGitRepo').mockReturnValue({
+    loading: false,
+    refetch: jest.fn(),
+    data: {
+      repoDetails: {
+        autoCommit: repoDetails.autoCommit || false,
+        branch: repoDetails.branch || 'main',
+        enabled: repoDetails.enabled ?? true,
+        filePath: repoDetails.filePath || '/flags.yaml',
+        repoIdentifier: repoDetails.repoIdentifier || 'harnesstest',
+        rootFolder: repoDetails.rootFolder || '/.harness/',
+        yamlError: repoDetails.yamlError || ''
+      },
+      repoSet: repoSet
+    }
+  } as any)
+}
+
+describe('useArchiveFlagDialog GitSync', () => {
+  test('it should open Git Modal if project is integrated with Git', async () => {
+    setUseGitRepoMock()
+
+    const customCommitMessage = 'MY COMMIT MESSAGE'
+    const archiveFlagMock = jest.fn()
+
+    setUseGitRepoMock({ autoCommit: false })
+
+    renderComponent({ archiveFlag: archiveFlagMock })
+
+    await userEvent.click(screen.getByRole('button', { name: openArchiveDialogBtn }))
+
+    await userEvent.type(screen.getByRole('textbox'), mockFeature.identifier)
+
+    await userEvent.click(screen.getByRole('button', { name: 'archive' }))
+
+    await waitFor(() => expect(screen.getByTestId('save-flag-to-git-modal-body')).toBeInTheDocument())
+
+    await userEvent.clear(screen.getByPlaceholderText('common.git.commitMessage'))
+    await userEvent.type(screen.getByPlaceholderText('common.git.commitMessage'), customCommitMessage)
+
+    expect(screen.getByPlaceholderText('common.git.commitMessage')).toHaveValue(customCommitMessage)
```

Comment:
We're getting the same element 3 times in a row. Might be worth getting it once and asserting thrice.
```suggestion
    const commitMessageElement = screen.getByPlaceholderText('common.git.commitMessage')
    
    await userEvent.clear(commitMessageElement)
    await userEvent.type(commitMessageElement, customCommitMessage)

    expect(commitMessageElement).toHaveValue(customCommitMessage)
```